src/utils/hex.cpp

```diff
@@ -92,6 +92,24 @@ int Hex2Bytes(const char *aHex, uint8_t *aBytes, uint16_t aBytesLength)
     return static_cast<int>(cur - aBytes);
 }
 
+size_t Uint2Hex(const uint64_t aUint, const uint8_t aUintLength, char *aHex)
```

Comment:
Can't `Long2Hex()` in this file fulfill your requirements?

Reply:
The issue with long2Hex is that the string in in little endian. I could write a string reversal if that would be preferable

Reply:
using `snprint()` for direct conversion gives compiler warnings that `%016lx` must be used for unsigned long but this then breaks 32bit architecture compatibility where it needs to be `%016llx`